networkit/cpp/graph/GraphBuilder.cpp

```diff
@@ -366,4 +378,97 @@ count GraphBuilder::numberOfEdges(const Graph &G) {
     }
 }
 
+// Parallel functions
+
+void GraphBuilder::addEdgeParallel(node a, node b) {
+    if (directed) {
+        addHalfOutEdgeParallel(Unsafe{}, HalfEdge{a, b});
+        addHalfInEdgeParallel(Unsafe{}, HalfEdge{b, a});
+    } else {
+        addHalfEdgeParallel(Unsafe{}, HalfEdge{a, b});
+        addHalfEdgeParallel(Unsafe{}, HalfEdge{b, a});
+    }
+}
+
+void GraphBuilder::addHalfEdgeParallel(Unsafe, HalfEdge halfEdge) {
+    node a = halfEdge.source;
+    node b = halfEdge.destination;
+    assert(a != b);
+    outEdgesPerThread[omp_get_thread_num()][a % omp_get_max_threads()].emplace_back(HalfEdge{a, b});
+}
+
+void GraphBuilder::addHalfInEdgeParallel(Unsafe, HalfEdge halfEdge) {
+    node a = halfEdge.source;
+    node b = halfEdge.destination;
+    assert(a != b);
+    inEdgesPerThread[omp_get_thread_num()][a % omp_get_max_threads()].emplace_back(HalfEdge{a, b});
+}
+
+void GraphBuilder::addHalfOutEdgeParallel(Unsafe, HalfEdge halfEdge) {
+    node a = halfEdge.source;
+    node b = halfEdge.destination;
+    assert(a != b);
+    outEdgesPerThread[omp_get_thread_num()][a % omp_get_max_threads()].emplace_back(HalfEdge{a, b});
+}
+
+void GraphBuilder::addEdgesToGraphParallel(Unsafe, Graph &G) {
```

Comment:
This function only gives valid results, if `addHalf/Out/In/EdgeParallel` was used for adding edges. Either make sure to check this accordingly or better change the public API during refactoring.